lib/iron_bank/local.rb

```diff
@@ -25,7 +25,9 @@ def first
 
     def where(conditions, limit: IronBank::Actions::Query::DEFAULT_ZUORA_LIMIT)
       records = store.values.select do |record|
-        conditions.all? { |field, value| record.public_send(field) == value }
+        conditions.all? do |field, match_value|
```

Comment:
In `QueryBuilder`, which is used by `Queryable` module, we have a `#restrict_single_range` method invoked as a validation step:

https://github.com/zendesk/iron_bank/blob/7aaa80c3280701faae5cbadef695842663760f38/lib/iron_bank/query_builder.rb#L65-L70

I looked at the Git history to understand why we put this restriction in place. I believe it's because, at the time, Zuora had poor support of `and/or` prioritization, but this doesn't appear to be the case anymore, as documented in [Filter Statements (ZOQL)](https://knowledgecenter.zuora.com/Central_Platform/Query/ZOQL/Filter_Statements).

My issue is that if `super` ends up being called from the `Local` module with this change, it might raises an error if the `conditions` contains a range and another values, e.g.:

```rb
{
  id: '123',
  type: ['foo', 'bar']
}
```

I opened https://github.com/zendesk/iron_bank/issues/64 because I'd rather see us removing the validation, and leveraging multiple fields query. So I'm OK to merge this into `Local` module.